src/lexer.py

```diff
@@ -0,0 +1,1060 @@
+#######################################
+# IMPORTS
+#######################################
+
+from strings_with_arrows import *
+
+import string
+
+#######################################
+# CONSTANTS
+#######################################
+
+DIGITS = '0123456789'
+LETTERS = string.ascii_letters
+LETTERS_DIGITS = LETTERS + DIGITS
+
+#######################################
+# ERRORS

```

Comment:
If this isn't auto-generated, some comments/documentation on all the methods would greatly help to understand them.